lib/Driver/ToolChains.cpp

```diff
@@ -522,6 +522,11 @@ ToolChain::constructInvocation(const CompileJobAction &job,
     Arguments.push_back("-track-system-dependencies");
   }
 
+  if (auto arg = context.Args.getLastArg(options::OPT_lto)) {
```

Comment:
Can we read this information out of `context.OI.LTOVariant` instead of looking for `OPT_lto` again?

Also, would it make sense to do this in `addCommonFrontendArgs()` instead? (If other calls to the frontend, like merge-modules and PCH building, ought to know if we're building for LTO, then it probably should be.)